### PR TITLE
Add API docs for owners' template resource.

### PR DESCRIPTION
**Pull request by martinrrm**
Service Index: https://github.com/NuGet/NuGet.Services.Index/pull/235

### CODE REVIEW

File: docs/TOC.md [hunk 1]
```diff
@@ -125,6 +125,7 @@
 #### [Catalog](api/catalog-resource.md)
 #### [Package content](api/package-base-address-resource.md)
 #### [Package details URL](api/package-details-template-resource.md)
+#### [Owner URL](api/package-owner-template-resource.md)
```
Comment (nkolev92) [inline docs/TOC.md:128]: I think these are sorted alphabetically.

File: docs/api/owner-details-template-resource.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
+---
```
Comment (nkolev92) [inline docs/api/owner-details-template-resource.md:1]: This doc LGTM.

**Review by nkolev92 — APPROVED** [review]

File: docs/api/owner-details-template-resource.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+The value of the `@id` in the service index is a URL string containing any of the following placeholder tokens:
+
+### URL placeholders
+
+Name        | Type    | Required | Notes
+----------- | ------- | -------- | -----
+`{owner}`   | string  | no       | The owner ID to get details for
+
+The server should accept `{owner}` value with any casing.
```
Comment (joelverhagen) [inline docs/api/owner-details-template-resource.md:55]: Consider this alternative:

> The casing requirements of `{owner}` should be defined by the package source. The client should retain the casing of the username in whatever form it was found in other API responses or in the package repository signature.
> 
> NuGet.org treats owner usernames in a case insensitive manner but not all package source may have this flexibility.

Today we are strict on client side for owner user name for client trust policy. So it would be worth clarifying the nuance. This is in contrast to package ID and version which are expected in many places to be case insensitive.

**Review by joelverhagen — CHANGES_REQUESTED** [review]
Owner usernames are not asserted to be case insensitive ecosystem wide at this time.

File: docs/api/owner-details-template-resource.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+ms.date: 4/15/2024
+ms.topic: reference
+ms.reviewer:
+---
+
+# Owner URL template
+
+It is possible for a client to build a URL that can be used by the user to see owner's page in their web
+browser. This is useful when a package source wants to show additional information about an owner that may not fit
+within the scope of what the NuGet client application shows.
```
[the remark this answers is not in the document]
Comment (martinrrm) [reply docs/api/owner-details-template-resource.md:15]: Not sure how it got formatted like that, I'll change every sentence to be one per line.

File: docs/api/owner-details-template-resource.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+The following `@type` values are used:
+
+@type value                     | Notes
+------------------------------- | -----
+OwnerDetailsUriTemplate/6.11.0  | The initial release
+
+## URL template
+
+The URL for the following API is the value of the `@id` property associated with one of the aforementioned
+resource `@type` values.
```
[the remark this answers is not in the document]
Comment (martinrrm) [reply docs/api/owner-details-template-resource.md:31]: Tbh I just copied from other documents that have the same section. @joelverhagen I don't see a lot of value in having this section in the documentation, do you think we should keep it? If so, how can we improve this section?

[the remark this answers is not in the document]
Comment (joelverhagen) [reply docs/api/owner-details-template-resource.md]: I don't feel strongly. Feel free to edit it according to @zivkan's suggestions. Sounds reasonable.

File: docs/api/owner-details-template-resource.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+NuGet.org treats owner usernames in a case insensitive manner but not all package source may have this flexibility.
+
+For example, nuget.org's package details template looks like this:
+
+```http
+https://www.nuget.org/profile/{owner}
+```
+
+If the client implementation needs to display a link to the package details for NuGet.Versioning 4.3.0, it would produce the following URL and provide it to the user:
```
Comment (donnie-msft) [inline docs/api/owner-details-template-resource.md:62]: This example appears to be from the package details template.

File: docs/api/owner-details-template-resource.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+ms.author: mruizmares
+ms.date: 4/15/2024
+ms.topic: reference
+ms.reviewer:
+---
+
+# Owner URL template
+
+It is possible for a client to build a URL that can be used by the user to see owner's page in their web browser. 
+This is useful when a package source wants to show additional information about an owner that may not fit within the scope of what the NuGet client application shows.
```
Comment (donnie-msft) [inline docs/api/owner-details-template-resource.md:14]: I'm not sure... maybe scratch this for now? As far as I know, nuget.org's owner profile pages are just a profile photo and listings of packages owned by that individual. They don't really add details... or can they?

From my spec's perspective, the purpose is to get some reassurance that the owner is really the owner I expect, by looking at their other published packages on this package source.

File: docs/api/owner-details-template-resource.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+The URL must be an absolute URL and the scheme (protocol) must be HTTPS.
+
+The value of the `@id` in the service index is a URL string containing any of the following placeholder tokens:
+
+### URL placeholders
+
+Name        | Type    | Required | Notes
+----------- | ------- | -------- | -----
+`{owner}`   | string  | no       | The owner ID to get details for
```
[the remark this answers is not in the document]
Comment (joelverhagen) [reply docs/api/owner-details-template-resource.md:49]: There are two kinds of required that matter I think: 1) the client rejects or errors out when no `{owner}` parameter is found and 2) we say it's required in the docs because we think it should really be there, but we make no effort to enforce it in the client. 

I think having the same URL for every owner (i.e. no `{owner}` token) is a bad user experience but I don't know how to best surface that error in client.

In short, I think saying required = yes in the docs as a description of intent is okay even if client does no validation on this. required = no gives the wrong intent

[the remark this answers is not in the document]
Comment (donnie-msft) [reply docs/api/owner-details-template-resource.md:49]: "Required yes" sounds good to me. 

I agree that it's a bad user experience to have identical links for all owners, but we could leave that up to the package source. I wanted to know whether to add validation in Client in my work in progress. I will not and just allow it to get the same repeated URL for each owner.
